### PR TITLE
Add GitHub Action workflow for CI

### PR DESCRIPTION
**Pull request by theseanything**
This is an initial workflow to enable CI using GitHub Actions. It makes use of
reusable workflows and composite actions defined in govuk-infrastructure. The
CI workflow is initially only used to support deployments to the new EKS
infrastructure and not to immediately replace the CI workflow in Jenkins.

The CI workflow makes use of individual jobs to run each testing and linting
tool. This allows the processes to run concurrently and improves the visibility
of status in the GitHub UI.

Most jobs make use of reusable workflows as they are consistent across the
majority of our repositories - this allows us to easily maintain them in a
single place. However, the test-ruby job (and/or integration-tests) is defined
per repo as backing services and other dependencies vary greatly between repos.
Backing services (such as databases) are created using a composite action
instead of a job service. This is because they contain a lot of shared config
and makes it easier for us to maintain.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Checkout Publishing API (for Content Schemas)
+        uses: actions/checkout@v3
+        with:
+          repository: alphagov/publishing-api
+          ref: deployed-to-production
+          path: vendor/publishing-api
```
Comment (kevindew) [inline .github/workflows/ci.yaml:41]: Just a quick drive-by comment. This won't be needed for this repo :)

Comment (theseanything) [reply .github/workflows/ci.yaml:41]: Ah good catch!

**Comment by theseanything** [comment]
Publishing API has a bunch of after test tasks including running pact tests that need to be incorporated.

**Review by marcpomfret — APPROVED** [review]
